client/sites-dashboard/components/wpcom-smp-commands.tsx

```diff
@@ -105,6 +105,35 @@ export const useCommandsArrayWpcom = ( {
 	const commandNavigation = useCommandNavigation( { navigate, currentRoute } );
 	const dispatch = useDispatch();
 
+	const navigateWithinSamePage = (
+		targetPath: string,
+		elementId: string,
+		commandParams: Pick< CommandCallBackParams, 'close' | 'command' >
+	) => {
+		const currentPath = window.location.pathname;
+		const targetUrl = new URL( targetPath, window.location.origin );
+
+		// Check if the user is on the same page but is not looking at the right section
+		// If not, scroll to the right section
+		if ( currentPath === targetUrl.pathname && window.location.hash !== `#${ elementId }` ) {
+			//Offset by the height of the navigation header from the top of the page
+			const fixedHeaderHeight = 72;
+			const element = document.getElementById( elementId );
+			if ( element ) {
+				const elementPosition = element.getBoundingClientRect().top + window.scrollY;
+				const offsetPosition = elementPosition - fixedHeaderHeight;
+
+				window.location.hash = elementId;
+				commandParams.close();
+				window.scrollTo( {
+					top: offsetPosition,
+					behavior: 'smooth',
+				} );
+			}
+		} else {
+			commandNavigation( targetUrl.href )( commandParams );
```

Comment:
Regarding this, I am only using `commandNavigation` for when you actually have to navigate to this page and not passing the page jumps through it. `useCommandNavigation` is designed to detect when you are specifically navigating to a different page and dispatch tracks event in that case so I opted to exclude the page jumps from there. For cases like this, when the user navigates within the same page, we can use the data from [`calypso_hosting_command_palette_command_select`](https://mc.a8c.com/tracks/live/?eventname=calypso_hosting_command_palette_command_select&user=&useragent=) event. 

Happy to reconsider this though if we want to consider jumping the same page towards `calypso_hosting_command_palette_navigate` event.